### ISSUE TITLE
Compatibility with Android 7 Nougat

### ISSUE THREAD

**Issue by rodrigograca31** [vstirbu/InstagramPlugin#85]
Hi everyone!

I noticed this plugin is not working in Android 7 ...
I get this error on ADB:
```
08-17 17:07:58.329 28925 28987 E PluginManager: Uncaught exception from plugin
08-17 17:07:58.329 28925 28987 E PluginManager: android.os.FileUriExposedException: file:///storage/emulated/0/Android/data/[PACKAGE]/files/instagram447377252.png exposed beyond app through ClipData.Item.getUri()
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.os.StrictMode.onFileUriExposed(StrictMode.java:1799)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.net.Uri.checkFileUriExposed(Uri.java:2346)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.content.ClipData.prepareToLeaveProcess(ClipData.java:845)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.content.Intent.prepareToLeaveProcess(Intent.java:8941)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.content.Intent.prepareToLeaveProcess(Intent.java:8926)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.app.Instrumentation.execStartActivity(Instrumentation.java:1690)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.app.Activity.startActivityForResult(Activity.java:4291)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.CordovaActivity.startActivityForResult(CordovaActivity.java:342)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.app.Activity.startActivityForResult(Activity.java:4249)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.CordovaInterfaceImpl.startActivityForResult(CordovaInterfaceImpl.java:66)
08-17 17:07:58.329 28925 28987 E PluginManager:         at com.vladstirbu.cordova.CDVInstagramPlugin.share(CDVInstagramPlugin.java:127)
08-17 17:07:58.329 28925 28987 E PluginManager:         at com.vladstirbu.cordova.CDVInstagramPlugin.execute(CDVInstagramPlugin.java:73)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.CordovaPlugin.execute(CordovaPlugin.java:98)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.PluginManager.exec(PluginManager.java:132)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.CordovaBridge.jsExec(CordovaBridge.java:57)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.apache.cordova.engine.SystemExposedJsApi.exec(SystemExposedJsApi.java:41)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.chromium.base.SystemMessageHandler.nativeDoRunLoopOnce(Native Method)
08-17 17:07:58.329 28925 28987 E PluginManager:         at org.chromium.base.SystemMessageHandler.handleMessage(SystemMessageHandler.java:7)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.os.Handler.dispatchMessage(Handler.java:102)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.os.Looper.loop(Looper.java:154)
08-17 17:07:58.329 28925 28987 E PluginManager:         at android.os.HandlerThread.run(HandlerThread.java:61)
```

I would appreciate if someone could take care of it (without breaking other Android versions...)
Thanks

**Comment by rodrigograca31** [comment]
For who solves this first there is a $50 bounty on it created by me:
https://www.bountysource.com/issues/48463381-compatibility-with-android-7-nougat

**Comment by moritzgloeckl** [comment]
Hi,

can you share some code or a guide on how to reproduce this? The plugin is working fine for me with a clean cordova app, canvasId as well as dataURL on a Samsung Galaxy S7 Edge running 7.0

Thanks!

**Comment by rodrigograca31** [comment]
oh boy..... seems was fixed in "ebba2d1fd1a590a967066fa27747fa6952b24232"

why didnt I check if there was updates to the extension? sorry....  let me test and make sure works for me....
